File: Java/src/main/java/com/gildedrose/ItemFactory.java
package com.gildedrose;

public class ItemFactory {

    public static Item createItem(String name, int sellIn, int quality) {
        if ("Aged Brie".equals(name)) {
            return new AgedBrieItem(name, sellIn, quality);
        } else if ("Backstage passes to a TAFKAL80ETC concert".equals(name)) {
            return new BackstageItem(name, sellIn, quality);
        } else if ("Sulfuras, Hand of Ragnaros".equals(name)) {
            return new SulfurasItem(name, sellIn, quality);
        }
        return new DefaultItem(name, sellIn, quality);
    }
}

File: Kotlin/src/main/kotlin/com/gildedrose/ItemFactory.kt
package com.gildedrose

fun createItem(name: String, sellIn: Int, quality: Int): Item {
    return when (name) {
        "Aged Brie" -> AgedBrieItem(name, sellIn, quality)
        "Backstage passes to a TAFKAL80ETC concert" -> BackstageItem(name, sellIn, quality)
        "Sulfuras, Hand of Ragnaros" -> SulfurasItem(name, sellIn, quality)
        else -> DefaultItem(name, sellIn, quality)
    }
}